lib/rules/selector-max-id/index.mjs

```diff
@@ -54,7 +55,8 @@ const rule = (primary, secondaryOptions) => {
 				if (
 					childNode.type === 'selector' ||
 					(isContextFunctionalPseudoClass(childNode) &&
-						!isIgnoredContextFunctionalPseudoClass(childNode))
+						!isIgnoredContextFunctionalPseudoClass(childNode)) ||
+					(childNode.type === 'pseudo' && isCheckedContextFunctionalPseudoClasses(childNode))
```

Comment:
```js
childNode.type === 'selector' ||
isUnignoredContextFunctionalPseudoClass(childNode) ||
isCheckedContextFunctionalPseudoClass(childNode)
```

Reply:
If we respect the following notice in the rule's README,

> This option has a higher precedence than `ignoreContextFunctionalPseudoClasses`.

we can call `isCheckedContextFunctionalPseudoClass()` before `isUnignoredContextFunctionalPseudoClass` like this:

```js
childNode.type === 'selector' ||
isCheckedContextFunctionalPseudoClass(childNode) ||
isUnignoredContextFunctionalPseudoClass(childNode)
```